Common/Product/SharedProject/UIThread.cs

```diff
@@ -26,7 +26,12 @@ class UIThread : UIThreadBase {
         private readonly Thread _uiThread;
 
         private UIThread() {
-            _scheduler = TaskScheduler.FromCurrentSynchronizationContext();
```

Comment:
I believe we could instead try checking `SyncronizationContext.Current` hear instead of catching an exception from `TaskScheduler`. That may be more clear and scope the fix better.